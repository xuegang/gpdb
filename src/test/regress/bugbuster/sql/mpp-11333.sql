DROP TABLE test_trig;
DROP FUNCTION fn_trig();
CREATE TABLE test_trig(id int, aaa text) DISTRIBUTED BY (id);
CREATE OR REPLACE FUNCTION fn_trig() RETURNS TRIGGER LANGUAGE plpgsql NO SQL AS $$
BEGIN
   NEW.id = NEW.id + 1;
   RAISE NOTICE '%', NEW.id;
RETURN NEW;
END
$$;
INSERT INTO test_trig VALUES (1, 'before creating trigger');
SELECT gp_segment_id, * FROM test_trig;
CREATE TRIGGER test_trig_1
BEFORE INSERT OR UPDATE ON test_trig
FOR EACH ROW
    EXECUTE PROCEDURE fn_trig();
INSERT INTO test_trig VALUES (0, 'after creating trigger');
--start_ignore
SELECT gp_segment_id, * FROM test_trig;
--end_ignore
