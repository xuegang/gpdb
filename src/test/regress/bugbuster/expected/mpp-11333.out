DROP TABLE test_trig;
ERROR:  table "test_trig" does not exist
DROP FUNCTION fn_trig();
ERROR:  function fn_trig() does not exist
CREATE TABLE test_trig(id int, aaa text) DISTRIBUTED BY (id);
CREATE OR REPLACE FUNCTION fn_trig() RETURNS TRIGGER LANGUAGE plpgsql NO SQL AS $$
BEGIN
   NEW.id = NEW.id + 1;
   RAISE NOTICE '%', NEW.id;
RETURN NEW;
END
$$;
INSERT INTO test_trig VALUES (1, 'before creating trigger');
SELECT gp_segment_id, * FROM test_trig;
 gp_segment_id | id |           aaa           
---------------+----+-------------------------
             0 |  1 | before creating trigger
(1 row)

CREATE TRIGGER test_trig_1
BEFORE INSERT OR UPDATE ON test_trig
FOR EACH ROW
    EXECUTE PROCEDURE fn_trig();
INSERT INTO test_trig VALUES (0, 'after creating trigger');
NOTICE:  1  (seg1 sdw11:18507 pid=1372)
--start_ignore
SELECT gp_segment_id, * FROM test_trig;
 gp_segment_id | id |           aaa           
---------------+----+-------------------------
             1 |  1 | after creating trigger
             0 |  1 | before creating trigger
(2 rows)

--end_ignore
